avocado/utils/lv_utils.py

```diff
@@ -0,0 +1,517 @@
+:param ramdisk_sparse_filename: Name of the ramdisk sparse file.
+:param disk: Name of the disk to create volume group. If not provided,
+             creates loop disk and creates volume group on it.
+
+Sample ramdisk params:
+- ramdisk_vg_size = "40000"
+- ramdisk_basedir = "/tmp"
+- ramdisk_sparse_filename = "virtual_hdd"
+
+Sample general params:
+- vg_name='autotest_vg',
+- lv_name='autotest_lv',
+- lv_size='1G',
+- lv_snapshot_name='autotest_sn',
+- lv_snapshot_size='1G'
+The ramdisk volume group size is in MB.
+"""
+
+import logging
+import os
+import re
+import shutil
+import time
+from . import process
+
+LOGGER = logging.getLogger('avocado.test')
+
+
+def get_diskspace(disk):
+    """
+    Get the entire disk space of a given disk
+    :return: size in bytes
+    """
+
+    result = process.system_output('fdisk -l %s' % disk, shell=True)
+    results = result.splitlines()
+    for line in results:
+        if line.startswith('Disk ' + disk):
+            pattern = re.compile(r", (.*?) bytes")
+            space = pattern.findall(line)[0]
+    return space
+
+
+def vg_ramdisk(disk, vg_name, ramdisk_vg_size,
+               ramdisk_basedir, ramdisk_sparse_filename):
+    """
+    Create vg on top of ram memory to speed up lv performance.
+    """
+    vg_size = ramdisk_vg_size
+    vg_ramdisk_dir = os.path.join(ramdisk_basedir, vg_name)
+    ramdisk_filename = os.path.join(vg_ramdisk_dir,
+                                    ramdisk_sparse_filename)
+    vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir, vg_name)
+    result = ""
+    if not os.path.exists(vg_ramdisk_dir):
+        os.mkdir(vg_ramdisk_dir)
+    try:
+        LOGGER.debug("Mounting tmpfs")
+        result = process.run("mount -t tmpfs tmpfs %s" %
+                             vg_ramdisk_dir, shell=True)
+
+        LOGGER.debug("Converting and copying /dev/zero")
+        if disk:
+            vg_size = get_diskspace(disk)
+        cmd = ("dd if=/dev/zero of=%s bs=1M count=1 seek=%s" %
+               (ramdisk_filename, vg_size))
+        result = process.run(cmd, shell=True)
+        if not disk:
+            LOGGER.debug("Finding free loop device")
+            result = process.run("losetup --find", shell=True)
+    except process.CmdError, ex:
+        LOGGER.error(ex)
+        vg_ramdisk_cleanup(ramdisk_filename, vg_ramdisk_dir, vg_name)
+        raise ex
+
+    if not disk:
+        loop_device = result.stdout.rstrip()
+    else:
+        loop_device = disk
+    try:
+        if not disk:
+            LOGGER.debug("Creating loop device")
+            result = process.run("losetup %s %s" %
+                                 (loop_device, ramdisk_filename), shell=True)
+        LOGGER.debug("Creating physical volume %s", loop_device)
+        result = process.run("pvcreate %s" % loop_device, shell=True)
+        LOGGER.debug("Creating volume group %s", vg_name)
+        result = process.run("vgcreate %s %s" %
```

Comment:
`result` is only used in log, but it's already logged by `process.run` so I believe you can drop it...